app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala

```diff
@@ -205,12 +295,60 @@ class RepresenteeController @Inject() (
             if (answers.fold(_.id, c => Some(c.id)).contains(id))
               Redirect(routes.RepresenteeController.checkYourAnswers())
             else {
+              val name: Option[IndividualName] = answers.fold(i => i.name, c => Some(c.name))
+              val response: EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]] =
+                id match {
+                  case RepresenteeNino(ninoId) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Right(ninoId), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeSautr(sautr) =>
+                    businessPartnerRecordService
+                      .getBusinessPartnerRecord(IndividualBusinessPartnerRecordRequest(Left(sautr), name))
+                      .map(bpr => Some(Left(bpr)))
+                  case RepresenteeCgtReference(cgtReference) => {
+                    subscriptionService.getSubscribedDetails(cgtReference).map(sd => Some(Right(sd)))
+                  }
+                  case NoReferenceId =>
+                    EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]](
+                      Future.successful(Right(None))
+                    )
+                  case _ =>
+                    EitherT[Future, Error, Option[Either[BusinessPartnerRecordResponse, SubscribedDetails]]](
+                      Future.successful(Left(Error("error")))
+                    )
+                }
+
+              val nameCheck: EitherT[Future, Error, Unit] = response.subflatMap(value =>
+                value match {
+                  case Some(Left(BusinessPartnerRecordResponse(_, _))) =>
+                    Right()
+                  case Some(Right(SubscribedDetails(Right(i), _, _, _, _, _, _)))
+                      if name.fold(false)(doNamesMatch(_, i)) =>
+                    Right()
+                  case Some(Right(SubscribedDetails(Left(_), _, _, _, _, _, _))) =>
```

Comment:
this is an error case if we get a trust name

Reply:
Done.